qip-0007.md

```diff
@@ -0,0 +1,105 @@
+```
+  QIP: 7
+  Layer: Consensus (hard fork)
+  Title: Sharded ledger
+  Author: wizeguyy <wizeguyy+qip@quai.org>
+  Comments-Summary: No comments yet.
+  Comments-URI: https://github.com/quainetwork/qips/wiki/Comments:QIP-0007
+  Status: Draft
+  Type: Standards Track
+  Created: 2023-12-05
+  License: BSD-2-Clause
+```
+
+## Abstract
+This specification proposes a UTXO ledger for one of the two tokens in the Quai two-token system.
+
+## Motivation
+The Quai protocol is designed to be a two-token system. The first token ledger is designed to be highly programmable, including a smart contract VM and an account based ledger to compliment the VM. The second token ledger, will sacrifice programability in order to achieve maximum computational and communication efficiency. The second token ledger will use a UTXO accounting system, due to its improved resource efficiency. This specification describes the design of that ledger.
+
+## Specification
+The design of the UTXO ledger requires consideration of the following details:
+* What is the structure of a transaction output?
+* What is the structure of a UTXO transaction?
+* How are those transactions recorded in a block?
+
+These questions will be answered in the following sections.
+
+### Transaction Output Structure
+The structure of a transaction output is as follows:
+| bits[0..4] | bits[4..8]   | bits[8..16] | bits[16..]             |
+|------------|--------------|-------------|------------------------|
+| version    | denomination | len(pubkey) | address (variable size) |
+
+#### Fixed Denomination Values
+The value field of a transaction output is constrained to 1 of 16 denominations, where each denomination represents a quantity of Qu. The denominations are given below:
+| denomination | number of Qu |
+|--------------|--------------|
+| 0            | 0.005 Qu     |
+| 1            | 0.01 Qu      |
+| 2            | 0.05 Qu      |
+| 3            | 0.1 Qu       |
+| 4            | 0.5 Qu       |
+| 5            | 1 Qu         |
+| 6            | 5 Qu         |
+| 7            | 10 Qu        |
+| 8            | 50 Qu        |
+| 9            | 100 Qu       |
+| 10           | 500 Qu       |
+| 11           | 1000 Qu      |
+| 12           | 5000 Qu      |
+| 13           | 10000 Qu     |
+| 14           | 50000 Qu     |
+| 15           | 100000 Qu    |
+
+### Transaction Structure
+Transactions will contain the following data, RLP encoded:
+```
+{
+  inputs: [
+    {
+      txhash: []byte,
+      index: uint8,
```

Comment:
what does index mean here? I dont see an explanation

Reply:
I think its a good idea to explain the input struct as well, similar to the outputs